### ISSUE TITLE
Broken links cause failure in wkhtmltopdf

### ISSUE THREAD

**Issue by bobintetley** [bobintetley/asm3#811]
It's absolutely ridiculous, but there is no way to tell wkhtmltopdf to ignore broken image links. I've looked into this many times and very thoroughly. 

Worse, if you feed it a document with broken image links, it will fail with an error and not do the PDF generation.

Add a new util method to parse an HTML document and verify that the image links are all absolute and valid before converting to PDF.  If it finds any that are not absolute links and they are not valid, it blanks the src attribute of the img so that wkhtmltopdf does not fail.

This would stop quite a few support calls.

**Comment by bobintetley** [comment]
after some experimenting, this does not apply to v0.12.5-1 that ships with Debian Buster and is currently deployed on us16/us17.

**Comment by bobintetley** [comment]
This is a bad solution and has been reverted. It could allow a malicious user to make us attack another server by filling a document with image links. Instead, the issue that prompted this problem was a user copying and pasting image?mode=media URLs from the media tab. I added support for those instead and rewrote the horrible links code to use HTMLParser instead of replacing strings to make it more robust. It also strips anything invalid at the same time.
